src/software/simulated_tests/mock_ai_wrapper.cpp

```diff
@@ -0,0 +1,9 @@
+#include "software/simulated_tests/mock_ai_wrapper.h"
+
+void MockAIWrapper::onValueReceived(World world)
+{
+    auto primitives     = std::vector<std::unique_ptr<Primitive>>();
+    auto primitives_ptr = std::make_shared<const std::vector<std::unique_ptr<Primitive>>>(
```

Comment:
Nit: can we use ConstPrimitiveVectorPtr instead?

Reply:
assuming you mean for specifying the type rather than 'auto`. I unfortunately can't use it for construction. Changed